cmd/gitserver/internal/server_grpc.go

```diff
@@ -1393,8 +1435,9 @@ func (gs *grpcServer) FirstEverCommit(ctx context.Context, request *proto.FirstE
 		return nil, err
 	}
 
-	return &proto.FirstEverCommitResponse{
-		Commit: commit.ToProto(),
+	return &proto.BehindAheadResponse{
+		Behind: behindAhead.Behind,
+		Ahead:  behindAhead.Ahead,
 	}, nil
```

Comment:
I think this should work too

```suggestion
	return behindAhead.ToProto(), nil
```